Vault 46676 update rar page (#2862)
hashicorp/web-unified-docs: content/vault/v2.x/content/docs/concepts/native-ai-agent-support/rich-authorization-requests.mdx

The RAR constraint blocks the `metadata` parameter even though the policy
allows the parameter.

## Templated paths

RAR `path` fields support the same [identity template syntax](/vault/docs/concepts/policies#templated-policies)
as ACL policy paths. When Vault evaluates an `authorization_details` claim, it resolves
`{{...}}` placeholders in each `path` using the authenticated entity's identity

> information.

Reviewer: ```suggestion
as ACL policy paths. When Vault evaluates an `authorization_details` claim, it resolves the
`{{...}}` placeholders in each `path` using the identity information for the authenticated entity identity
```
Style correction: avoid possessives

RAR paths support all template parameters available in ACL policies. Refer to the
[templated policies parameter reference](/vault/docs/concepts/policies#parameters)
for the full list.

<Note>

Using wildcards (`+`) and globs (`*`) in the rendered output of identity templates will result in a "permission denied" response from Vault.

</Note>

<Note>

Use IDs wherever possible in templates. Each ID is unique and stable, whereas
names can change over time and be reused.

> </Note>

Reviewer: ```suggestion
Be aware of the following considerations when using templates:

- Vault returns a "permission denied" response when the rendered output of an
  identity template includes wildcards (`+`) or globs (`*`).

- Use IDs wherever possible. Each ID is unique and stable, whereas names can
  be reused and change over time.
```
Style correction: write in active voice, avoid possessives, do not use back-to-back asides

> **Example:**

Reviewer: ```suggestion
```

We don't need two introductions to the same example

The following `authorization_details` object restricts the JWT to the
authenticated entity's own path in the KVv2 secret engine, using the same

> `{{identity.entity.id}}` template you would use in an ACL policy:

Reviewer: ```suggestion
The following `authorization_details` object uses the same
`{{identity.entity.id}}` template you might use in an ACL policy to restrict
the JWT to the KVv2 plugin path associated with the authenticated entity:
```
Style correction: avoid possessives, use declarative language

```json
{
  "authorization_details": [
    {
      "type": "vault:path_access",
      "path": "secret/data/users/{{identity.entity.id}}",
      "capabilities": ["read"]
    }
  ]
}
```

> When Vault evaluates this constraint, it replaces `{{identity.entity.id}}` with

Reviewer: ```suggestion
When Vault evaluates the authorization constraint, it replaces `{{identity.entity.id}}` with
```
Style correction: avoid "this" as a pronoun

the actual entity ID (for example, `abc123`) and checks whether the request path
`secret/data/users/abc123` is permitted. The ACL policy must independently allow

> this path for the request to succeed.

Reviewer: ```suggestion
the entity access to the target path for the request to succeed.
```
Style correction: avoid "this" as a pronoun

### Unresolvable templates

> If a template placeholder cannot be resolved, Vault silently skips that `authorization_details`

Reviewer: ```suggestion
If Vault cannot resolve a template placeholder, it silently skips that `authorization_details`
```

entry. If no other entries match the request, Vault denies the request with a
`permission denied` error.

This means you can include a templated constraint alongside a static fallback

> without causing an error when the template is unresolvable:

Reviewer: ```suggestion
The fact that Vault skips unmatched placeholders means you can include a templated constraint alongside a static fallback
without causing an error when Vault cannot resolve the template. For example, in the following authorization constrain, Vault skips the first, unresolvable entry and evaluates the second entry normally:
```
Style correction: avoid "this" as a pronoun, write in active voice

> **Example:**

Reviewer: ```suggestion
```

```json
{
  "authorization_details": [
    {
      "type": "vault:path_access",
      "path": "secret/data/users/{{identity.entity.UNKNOWN}}",
      "capabilities": ["read"]
    },
    {
      "type": "vault:path_access",
      "path": "secret/data/users/fallback",
      "capabilities": ["read"]
    }
  ]
}
```

In this example, the first entry is skipped (unresolvable), and the second

> entry is evaluated normally.

Reviewer: ```suggestion
```
Folded into the  example introduction.

- Creating or updating at `pki/issue/my-role` (with `common_name` required)
- Reading or listing from `secret/data/app-*` paths (except version 1)

## Default behavior

Vault enables RAR by default, and OAuth 2.0 tokens with `authorization_details` always honor the RAR authentication details.

If the JWT does not contain an `authorization_details` claim, Vault rejects the request with an error indicating that the claim is required.

We recommend using RAR for all agentic workflows to enable fine-grained authorization constraints.

## Making RAR optional

> While not recommended, making RAR optional can be useful when your identity provider does not support RAR or during a migration period.

Reviewer: ```suggestion
While not recommended, making RAR optional can be useful during a migration period or when your identity provider does not support RAR.
```
Style correction: it it typically easier for folks to follow a sentence when the simpler phrase ("during a migration phase") comes first.

By default, Vault requires the `authorization_details` claim in OAuth JWTs used

> with the Agent Registry. You can make RAR optional in two ways:

Reviewer: Suggest modifying this introduction to include the modification information from below:
```
with the Agent Registry. You can make RAR optional in two way, and can modify the
settings at any time. The modified settings take effect on the next applicable
request:

```

1. **Per OAuth resource server profile**: Set `optional_authorization_details` to `true` on the
   [OAuth resource server configuration](/vault/api-docs/system/config-oauth-resource-server).

>    This applies to all AI agents authorized by that OAuth resource server.

Reviewer: ```suggestion
   The OAuth server configuration applies to all AI agents authorized by that OAuth resource server.
```

1. **Per agent registration**: Set `optional_authorization_details` to `true` on the
   [Agent Registry record](/vault/api-docs/secret/agent-registry).

>    This affects only the specific agent, provided the corresponding OAuth resource server profile has `optional_authorization_details` set to `false`.

Reviewer: ```suggestion
   Configuring the agent registry record only affects the specific agent unless the corresponding OAuth resource server profile has `optional_authorization_details` set to `false`.
```

> <Note>

Reviewer: ```suggestion
```'
Style correction: do not rely on asides/notes for text that should be part of the normal body of the doc

> These settings can be modified at any time and take effect on subsequent requests.

Reviewer: ```suggestion
```
Folded above

</Note>

> Vault provides two methods for making RAR optional so you can:

Reviewer: ```suggestion
Vault provides multiple configuration methods so you can:
```
